src/hotspot/share/oops/klass.cpp

```diff
@@ -873,12 +873,12 @@ void Klass::set_archived_java_mirror(int mirror_index) {
 
 void Klass::check_array_allocation_length(int length, int max_length, TRAPS) {
   if (length > max_length) {
-    if (!THREAD->in_retryable_allocation()) {
-      report_java_out_of_memory("Requested array size exceeds VM limit");
+    report_java_out_of_memory("Requested array size exceeds VM limit");
```

Comment:
Again not obvious this should now be unconditional

Reply:
Same reasoning as for `MemAllocator::Allocation::check_out_of_memory`.